module1-introduction-to-backend/r1.1-beginners-guide-nodejs-npm/README.md

```diff
@@ -0,0 +1,110 @@
+})
+```
+
+Using the http module, one of many useful Node.js libraries, an HTTP server is created. The server is set to listen on the specified host name and port. When the server is ready, the callback function is called, in this case informing us that the server is running. Whenever a new request is received from a client, the request event is called, providing two objects: a request and a response. The first provides details of the request, although not used in this example, and the second is used to return data to the client.
+
+### Some brief history first
+Node.js is only 12 years old, not a long time in tech. In comparison. JavaScript itself is more than 20 years old and we know that it is what runs in our browsers. JavaScript was created at Netscape, which also tried to create LiveWire, an attempt at server-side JavaScript. Unfortunately it wasn't very successful and server-side JavaScript did not grow in popularity until the introduction of Node.js in 2009.
+
+One key factor that led to the rise of Node.js was the timing. Just a few years earlier, JavaScript had started to be considered as a more serious language, thanks to "Web 2.0" applications (such as Flickr, Gmail, etc.) that showed the world what a modern experience on the web could be like. JavaScript engines also became considerably better as many browsers competed to offer users the best performance. Development teams behind major browsers worked hard to offer better support for JavaScript and find ways to make JavaScript run faster. The engine that Node.js uses under the hood, V8 (also known as Chrome V8 for being the open-source JavaScript engine of The Chromium Project), improved significantly due to this competition.
+
+When Ryan Dahl wrote Node.js, one of his motivations was the limited possibilities of the most popular web server in 2009, Apache HTTP Server, to handle a lot of concurrent connections and the most common way of creating code which either blocked the entire process or implied multiple execution stacks in the case of simultaneous connections. A Node.js app runs in a single process, without creating a new thread for every request. Node.js provides a set of asynchronous I/O primitives in its standard library that prevent JavaScript code from blocking and generally, libraries in Node.js are written using non-blocking paradigms, making blocking behavior the exception rather than the norm.
+
+### Features of Node.js
+Firstly, though `.js` is the standard filename extension for JavaScript code, the name "Node.js" doesn't refer to a particular file in this context and is merely the name of the product.
+
+1. Node.js is really fast: Having been built on Google Chrome's V8 JavaScript engine, its library is extremely fast for code execution.
+2. Node Package Manager (NPM): Node Package Manager has more than 50,000 bundles, so whatever functionality is required for an application can be easily imported from NPM.
```

Comment:
```suggestion
2. Rich libraries: Node Package Manager (NPM) has more than 50,000 bundles, so whatever functionality is required for an application can be easily imported from NPM.
```

the thing after the number should be the more abstract point while the specific example comes after (that's how 1, 3, 4, 5)